src/creator.rs

```diff
@@ -97,53 +93,244 @@ impl<H: Hasher> Creator<H> {
         }
     }
 
-    fn check_ready(&self) -> bool {
-        if self.current_round == 0 {
-            return true;
-        }
-        if self.current_round > self.max_round {
-            warn!(target: "AlephBFT-creator", "{:?} Maximum round reached. Not creating another unit.", self.node_ix);
-            return false;
+    async fn wait_until_ready(&mut self, round: Round) {
+
+        let prev_round_index = match round.checked_sub(1) {
+            Some(prev_round) => prev_round as usize,
+            None => return,
+        };
+
+        let delay = time::sleep((self.create_lag)(round.into()));
+        tokio::pin!(delay);
+        loop {
+            if usize::from(round) < self.n_candidates_by_round.len() - 1 {
+                return; // Since we get unit from round r, we have enough units from previous rounds to create our unit 
+            }
+
+            tokio::select! {
```

Comment:
I believe, we used to avoid having tokio as a dependency and used the `futures` crate for select and stuff. Did we change this?

Reply:
And actually, the code does not compile now because tokio is not added to dependencies.

Reply:
Done